src/Middleware/StaticFiles/src/StaticFilesEndpointRouteBuilderExtensions.cs

```diff
@@ -42,17 +48,12 @@ public static class StaticFilesEndpointRouteBuilderExtensions
         this IEndpointRouteBuilder endpoints,
         string filePath)
     {
-        if (endpoints == null)
-        {
-            throw new ArgumentNullException(nameof(endpoints));
-        }
+        ArgumentNullException.ThrowIfNull(endpoints);
+        ArgumentNullException.ThrowIfNull(filePath);
 
-        if (filePath == null)
-        {
-            throw new ArgumentNullException(nameof(filePath));
-        }
-
-        return endpoints.MapFallback(CreateRequestDelegate(endpoints, filePath));
+        return endpoints
+            .MapFallback(CreateRequestDelegate(endpoints, filePath))
+            .WithMetadata(new HttpMethodMetadata(_supportedHttpMethods));
```

Comment:
Do we know that this doesn't affect CORS?
See https://github.com/dotnet/aspnetcore/issues/42994 for an issue with CORS and HttpMethodMetadata